common/config/src/main/java/io/helidon/common/config/Config.java

```diff
@@ -0,0 +1,346 @@
+     * logging:
+     *      app.level = INFO
+     *      level = WARNING
+     * </pre>
+     * The {@code Config} instances {@code name1} and {@code name2} represents same data and
+     * in fact refer to the same object:
+     * <pre>
+     * Config name1 = config
+     *                  .get("app")
+     *                  .get("name");
+     * Config name2 = config
+     *                  .get("app")
+     *                  .detach()               //DETACHED node
+     *                  .get("name");
+     *
+     * assert name1.asString() == "Example 1";
+     * assert name2.asString() == "Example 1";  //DETACHED node
+     * </pre>
+     * The only difference is the key each node returns:
+     * <pre>
+     * assert name1.key() == "app.name";
+     * assert name2.key() == "name";            //DETACHED node
+     * </pre>
+     * <p>
+     * See asMap() for example of config detaching.
+     *
+     * @return returns detached Config instance of same config node
+     * @throws io.helidon.common.config.ConfigException if not defined
+     */
+    default Config detach() throws ConfigException {
+        throw new ConfigException("unsupported; do you have a valid config module in the classpath?");
+    }
+
+    /**
+     * Returns {@code true} if the node exists, whether an object, a list, a
+     * value node, etc.
+     *
+     * @return {@code true} if the node exists
+     */
+    boolean exists();
+
+    /**
+     * Returns {@code true} if this node exists and is a leaf node (has no
+     * children).
+     * <p>
+     * A leaf node has no nested configuration subtree and has a single value.
+     *
+     * @return {@code true} if the node is existing leaf node, {@code false}
+     *         otherwise.
+     */
+    boolean isLeaf();
+
+    /**
+     * Returns {@code true} if this node exists and is Type#Object.
+     *
+     * @return {@code true} if the node exists and is Type#Object, {@code false}
+     *         otherwise.
+     */
+    boolean isObject();
+
+    /**
+     * Returns {@code true} if this node exists and is Type#List.
+     *
+     * @return {@code true} if the node exists and is Type#List, {@code false}
+     *         otherwise.
+     */
+    boolean isList();
+
+    //
+    // accessors
+    //
+
+    /**
+     * Typed value as a {@link ConfigValue}.
+     *
+     * @param type type class
+     * @param <T>  type
+     * @return typed value
+     * @see ConfigValue#map(java.util.function.Function)
+     * @see ConfigValue#get()
+     * @see ConfigValue#orElse(Object)
+     */
+    <T> ConfigValue<T> as(Class<T> type);
+
+    // shortcut methods
+
+    /**
+     * Returns list of specified type.
+     *
+     * @param type type class
+     * @param <T>  type of list elements
+     * @return a typed list with values
+     * @throws io.helidon.common.config.ConfigException in case of problem to map property value.
+     */
+    default <T> ConfigValue<List<T>> asList(Class<T> type) throws ConfigException {
```

Comment:
This should not have a default implementation. Esp. not an implementation like this